BraintreeCore/src/main/java/com/braintreepayments/api/ApiClient.kt

```diff
@@ -35,21 +29,46 @@ class ApiClient constructor(private val braintreeClient: BraintreeClient) {
         try {
             val body = paymentMethod.buildJSON().toString()
             braintreeClient.sendPOST(url, body) { responseBody, httpError ->
-                if (responseBody != null) {
-                    try {
-                        callback.onResult(JSONObject(responseBody), null)
-                    } catch (exception: JSONException) {
-                        callback.onResult(null, exception)
-                    }
-                } else {
-                    callback.onResult(null, httpError)
-                }
+                parseResponse(responseBody, httpError, callback)
             }
         } catch (exception: JSONException) {
-            callback.onResult(null, exception)
+            returnError(callback, exception)
+        }
+    }
+
+    private fun parseResponse(
+        responseBody: String?,
+        exception: Exception?,
+        callback: TokenizeCallback,
+        analyticsEventName: String? = null) {
```

Comment:
Nitpick comment - I feel like sending analytics goes beyond the responsibility of the name of the function here `parseResponse`

Reply:
That's a good point. `parseResponseBodyToJSON()` would be a good alternative. And we can send the analytics events inline.